code/modules/clothing/chameleon.dm

```diff
@@ -257,6 +257,53 @@
 	. = ..()
 	chameleon_action.emp_randomise(INFINITY)
 
+// Traitor item "Doppelganger Suit", behaves the same way as the reactive stealth armor, but has the added bonus of chameleon disguise.
+/obj/item/clothing/suit/chameleon/doppel
+	var/active = FALSE
+	resistance_flags = LAVA_PROOF | FIRE_PROOF | ACID_PROOF
+
+/obj/item/clothing/suit/chameleon/doppel/proc/doppel_cooldown(mob/user)
+	cooldown = FALSE
+
+/obj/item/clothing/suit/chameleon/doppel/proc/doppel_cloaking(mob/user)
+	owner.alpha = initial(owner.alpha)
+
+/obj/item/clothing/suit/chameleon/doppel/attack_self(mob/user)
+	active = !active
+	if(active)
+		to_chat(user, "<span class='notice'>[src] is now active.</span>")
+	else
+		to_chat(user, "<span class='notice'>[src] is now inactive.</span>")
+		add_fingerprint(user)
+	user.update_inv_wear_suit()
+	for(var/X in actions)
+		var/datum/action/A = X
+		A.UpdateButtonIcon()
+
+/obj/item/clothing/suit/chameleon/doppel/emp_act(severity)
+	active = FALSE
+	cooldown = TRUE
+	addtimer(CALLBACK(src, .proc/doppel_cooldown), 400)
+	if(istype(loc, /mob/living/carbon/human))
+		var/mob/living/carbon/human/C = loc
+		C.update_inv_wear_suit()
+	..()
+
+/obj/item/clothing/suit/chameleon/doppel/hit_reaction(mob/living/carbon/human/owner, atom/movable/hitby, attack_text = "the attack", final_block_chance = 0, damage = 0, attack_type = MELEE_ATTACK)
+	if(!active | cooldown == TRUE)
+		return FALSE
+	else
+		var/mob/living/simple_animal/hostile/illusion/escape/E = new(owner.loc)
+		E.Copy_Parent(owner, 70)
+		E.GiveTarget(owner) //so it starts running right away
+		E.Goto(owner, E.move_to_delay, E.minimum_distance)
+		owner.alpha = 0
+		owner.visible_message("<span class='danger'>[owner] is hit by [attack_text] in the chest!</span>") //We pretend to be hit, since blocking it would stop the message otherwise
+		addtimer(CALLBACK(src, .proc/doppel_cloaking), 70)
```

Comment:
Since your `doppel_cloaking` proc doesn't know what `owner` is, you need to pass it an argument here in the callback:
```dm
addtimer(CALLBACK(src, .proc/doppel_cloaking, owner), 70)
```
And then with the cloaking proc:
```dm
/obj/item/clothing/suit/chameleon/doppel/proc/doppel_cloaking(mob/living/carbon/human/owner)
	owner.alpha = initial(owner.alpha)
```